commons/src/main/java/fr/free/nrw/commons/category/CategorizationFragment.java

```diff
@@ -367,8 +452,28 @@ private void startUpdatingCategoryList() {
         if (lastUpdater != null) {
             lastUpdater.cancel(true);
         }
+
+        if (methodAUpdater != null) {
+            methodAUpdater.cancel(true);
+        }
+
+
+        ArrayList<CategoryItem> itemList = new ArrayList<CategoryItem>(itemSet);
+
+        methodAUpdater = new MethodAUpdater();
         lastUpdater = new CategoriesUpdater();
+
         Utils.executeAsyncTask(lastUpdater, executor);
+        Utils.executeAsyncTask(methodAUpdater, executor);
+
+
+        categoriesAdapter.setItems(itemList);
```

Comment:
Sorry for the late response. We had to make some unplanned releases at work.

AsyncTask.doInBackground() is a function that executes asynchronously of the main / UI thread. You're back on the main / UI thread in the onPostExecute() function. If you want code to execute after the task completes, that typically goes in onPostExecute() itself. For instance, if you want to download an image, it might look something like:

UI thread:
  1 User taps button to download image.
  2 Button spawns AsyncTask in it's onClick listener.
  --- The app is responsive because the UI thread is not blocked. ---

Background thread:
  1 doInBackground() runs and the image is downloaded after some time.

UI thread:
  1 onPostExecute() is called on the UI thread with the image fully downloaded and available. The function just calls ImageView.setImage() and exits.
  --- Image is shown and the app is responsive because the UI thread is not blocked. ---

AsyncTask is probably the most infamous function in Android so there's piles of info on them. I personally loathe them and try to use libraries that abstract threading, such as Volley, when possible, but it's good to have some experience with them. There's probably a metric of Android code health measured in how few AsyncTasks are used.